### PR TITLE
User story 4

### LINKED ISSUE #11: Save boomarks at local storage
## Value Statement
As a User I want save my bookmarks of my favorite clothing so I can still see them when I reload my page.

## Mockup
 mockup-favorites

## Acceptance Criteria
- every change on the bookmark state should be saved in the localStorage
- reload of the page should load the last state of my Clothing cards.

## Tasks
- [x] Save bookmarked Clothing cards on localStorage
- [x] Rendering bookmarked Clothing cards when reloading page

## Complexity
- medium

### PR DESCRIPTION
**Pull request by stschuessler**
I've implemented `localStorage` and used an `if-else` condition to check if `localStorage` is already set. 
I also added a `isBookmark` prop with boolean value, replaced the useState and used the slice method to save the changes in localStorage.

This resolves #11 

------

Update: Implemented the handleBookmark function to handle the bookmark in the App component (compare review comment)

------

Update: Implemented review suggestions and implemented clothes state variable and extracted `getItem` and `setItem` into its own function

-----

Update: Implemented suggestions from Lukas and did some adjustments on the testing and storybook files

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,19 +1,31 @@
...
+      {data.map((item) => (
         <Clothing
-          key={data.id}
-          title={data.title}
-          color={data.color}
-          pattern={data.pattern}
-          material={data.material}
-          fitting={data.fitting}
-          imageUrl={data.imageUrl}
+          data={data}
```
Comment (StefanBehring) [inline src/App.js:19]: You are giving down the complete `data`. Do you need that in your Clothing component?

File: src/App.js [hunk 2]
```diff
@@ -1,19 +1,45 @@
 import Clothing from './components/Clothing'
 import styled from 'styled-components/macro'
-import data from './mockup-data'
+import mockupData from './mockup-data'
 
 function App() {
+  let data
```
Comment (macfrei) [inline src/App.js:6]: This should be a state 👍

File: src/components/Bookmark.js [hunk 3]
```diff
@@ -1,15 +1,18 @@
 import { useState } from 'react'
 import styled from 'styled-components/macro'
 
-function Bookmark() {
-  const [active, setActive] = useState(false)
+function Bookmark({ data, id, isBookmarked, onClickBookmark }) {
+  let bookmarkActive = data.find((item) => item.id === id)
```
Comment (macfrei) [inline src/components/Bookmark.js:5]:
```suggestion
  const bookmarkActive = data.find((item) => item.id === id)
```

File: src/App.js [hunk 4]
```diff
@@ -1,19 +1,45 @@
...
+
+    localStorage.setItem(
+      'localClothing',
+      JSON.stringify([
+        ...data.slice(0, index),
+        bookmarkActive,
+        ...data.slice(index + 1),
+      ]),
+    )
+  }
```
Comment (macfrei) [inline src/App.js:26]: I added a note in our course channel about how to implement `localStorage` with a bit better readability. Maybe have a look and change your implementation - if you want 🙂

Comment (macfrei) [inline src/App.js:26]: In your function `handleBookmark` you are currently only updating the localStorage and not the `data`. I would suggest making `data` a state and then updating the state here as well. How about naming the state `[clothes, setClothes]`?
```suggestion
  const handleBookmark = (bookmarkActive) => {
    const index = data.findIndex((card) => card.id === bookmarkActive.id)
    const newClothingArray = [
        ...clothes.slice(0, index),
        bookmarkActive,
        ...clothes.slice(index + 1),
      ]
      
    setClothes(newClothingArray)
    localStorage.setItem(
      'localClothing',
      JSON.stringify(newClothingArray),
    )
  }
```

Comment (stschuessler) [reply src/App.js]: It works 🙂 Thanks for inspiration!

**Pull request merged**